src/library_wasmfs_fetch.js

```diff
@@ -23,10 +26,20 @@ mergeInto(LibraryManager.library, {
         // the actual read below.
         return Promise.resolve();
       }
-
+      var fileUrl = wasmFS$backendFileNames[file] ? wasmFS$backendFileNames[file] : '';
       // This is the first time we want the file's data.
-      // TODO: real URL!
-      var url = 'data.dat';
+      var url = '';
+      if (isAbs) {
+        try {
+          var u = new URL(fileUrl, _baseUrl);
+          url = u.toString();
+        } catch (e) {
+        }
+      } else {
+        url = _baseUrl;
+      }
+      if (url === '')
+        url = 'data.dat';
```

Comment:
We probably don't need this part anymore. I think `data.dat` was just a placeholder value.